The job **`x86_64-gnu-llvm-21`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/27677557493/job/81856646829) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/81856646829) [(plain)](https://github.com/rust-lang/rust/commit/a130f53637e91de1ab53884220c5461859945510/checks/81856646829/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling rustc_public_bridge v0.0.0 (/checkout/compiler/rustc_public_bridge)
error[E0658]: use of unstable library feature `strip_circumfix`
   --> compiler/rustc_metadata/src/diagnostics.rs:269:44
    |
269 |             if let Some(libname) = libname.strip_circumfix("lib", ".a") {
    |                                            ^^^^^^^^^^^^^^^
    |
    = note: see issue #147946 <https://github.com/rust-lang/rust/issues/147946> for more information
    = help: add `#![feature(strip_circumfix)]` to the crate attributes to enable
    = note: this compiler was built on 2026-05-30; consider upgrading it if it is out of date
---
[RUSTC-TIMING] rustc_query_impl test:false 79.884
error[E0658]: use of unstable library feature `strip_circumfix`
    --> compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:1894:50
     |
1894 |                         && let Some(code) = code.strip_circumfix('\'', '\'')
     |                                                  ^^^^^^^^^^^^^^^
     |
     = note: see issue #147946 <https://github.com/rust-lang/rust/issues/147946> for more information
     = help: add `#![feature(strip_circumfix)]` to the crate attributes to enable
     = note: this compiler was built on 2026-05-30; consider upgrading it if it is out of date

error[E0658]: use of unstable library feature `strip_circumfix`
    --> compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:1911:50
     |
1911 |                         && let Some(code) = code.strip_circumfix('"', '"')
     |                                                  ^^^^^^^^^^^^^^^
     |
     = note: see issue #147946 <https://github.com/rust-lang/rust/issues/147946> for more information
     = help: add `#![feature(strip_circumfix)]` to the crate attributes to enable
     = note: this compiler was built on 2026-05-30; consider upgrading it if it is out of date
```

</details>

--- a/compiler/rustc_metadata/src/lib.rs
+++ b/compiler/rustc_metadata/src/lib.rs
@@ -1,6 +1,7 @@
 // tidy-alphabetical-start
 #![allow(internal_features)]
 #![cfg_attr(bootstrap, feature(result_option_map_or_default))]
+#![cfg_attr(bootstrap, feature(strip_circumfix))]
 #![feature(error_iter)]
 #![feature(file_buffered)]
 #![feature(gen_blocks)]
@@ -8,7 +9,6 @@
 #![feature(min_specialization)]
 #![feature(never_type)]
 #![feature(proc_macro_internals)]
-#![feature(strip_circumfix)]
 #![feature(trusted_len)]
 // tidy-alphabetical-end
 

--- a/compiler/rustc_trait_selection/src/lib.rs
+++ b/compiler/rustc_trait_selection/src/lib.rs
@@ -11,6 +11,7 @@
 //! This API is completely unstable and subject to change.
 
 // tidy-alphabetical-start
+#![cfg_attr(bootstrap, feature(strip_circumfix))]
 #![feature(associated_type_defaults)]
 #![feature(default_field_values)]
 #![feature(deref_patterns)]
@@ -18,7 +19,6 @@
 #![feature(iter_intersperse)]
 #![feature(iterator_try_reduce)]
 #![feature(never_type)]
-#![feature(strip_circumfix)]
 #![feature(try_blocks)]
 #![feature(unwrap_infallible)]
 #![feature(yeet_expr)]

--- a/library/core/src/slice/mod.rs
+++ b/library/core/src/slice/mod.rs
@@ -2744,8 +2744,6 @@ impl<T> [T] {
     /// # Examples
     ///
     /// ```
-    /// #![feature(strip_circumfix)]
-    ///
     /// let v = &[10, 50, 40, 30];
     /// assert_eq!(v.strip_circumfix(&[10], &[30]), Some(&[50, 40][..]));
     /// assert_eq!(v.strip_circumfix(&[10], &[40, 30]), Some(&[50][..]));
@@ -2756,7 +2754,7 @@ impl<T> [T] {
     /// assert_eq!(v.strip_circumfix(&[10, 50], &[]), Some(&[40, 30][..]));
     /// ```
     #[must_use = "returns the subslice without modifying the original"]
-    #[unstable(feature = "strip_circumfix", issue = "147946")]
+    #[stable(feature = "strip_circumfix", since = "CURRENT_RUSTC_VERSION")]
     pub fn strip_circumfix<S, P>(&self, prefix: &P, suffix: &S) -> Option<&[T]>
     where
         T: PartialEq,

--- a/library/core/src/str/mod.rs
+++ b/library/core/src/str/mod.rs
@@ -2510,15 +2510,13 @@ impl str {
     /// # Examples
     ///
     /// ```
-    /// #![feature(strip_circumfix)]
-    ///
     /// assert_eq!("bar:hello:foo".strip_circumfix("bar:", ":foo"), Some("hello"));
     /// assert_eq!("bar:foo".strip_circumfix("foo", "foo"), None);
     /// assert_eq!("foo:bar;".strip_circumfix("foo:", ';'), Some("bar"));
     /// ```
     #[must_use = "this returns the remaining substring as a new slice, \
                   without modifying the original"]
-    #[unstable(feature = "strip_circumfix", issue = "147946")]
+    #[stable(feature = "strip_circumfix", since = "CURRENT_RUSTC_VERSION")]
     pub fn strip_circumfix<P: Pattern, S: Pattern>(&self, prefix: P, suffix: S) -> Option<&str>
     where
         for<'a> S::Searcher<'a>: ReverseSearcher<'a>,

